### PR TITLE
Run tests for Plus images

### PR DESCRIPTION
**Pull request by lucacome**
Test the NGINX Plus images in the pipeline

### CODE REVIEW

File: .github/actions/smoke-tests/action.yaml [hunk 1]
```diff
@@ -43,10 +59,13 @@ runs:
         cache-from: type=gha,scope=${{ inputs.image }}
         cache-to: type=gha,scope=${{ inputs.image }},mode=max
         target: goreleaser
-        tags: 'docker.io/nginx/nginx-ingress:${{ inputs.image }}-${{ github.sha }}'
+        tags: 'docker.io/nginx/${{ steps.ingress-type.outputs.name }}:${{ inputs.image }}-${{ github.sha }}'
         load: true
         pull: true
         build-args: BUILD_OS=${{ inputs.image }}
+        secrets: |
+          "nginx-repo.crt=${{ inputs.nginx-crt }}"
```
Comment (soneillf5) [inline .github/actions/smoke-tests/action.yaml:67]: where do these certs come from ?

File: .github/actions/smoke-tests/action.yaml [hunk 2]
```diff
@@ -32,6 +38,16 @@ runs:
         path: ${{ github.workspace }}/dist
         key: nginx-ingress-${{ github.run_id }}-${{ github.run_number }}-single
 
+    - name: Ingress type
+      id: ingress-type
+      run: |
+        plus=""
+        if [[ ${{ inputs.image }} == *plus* ]]; then
```
Comment (soneillf5) [inline .github/actions/smoke-tests/action.yaml:45]: I'm still not 100% sure about making decisions about an image based on if the string 'plus' is in the name. @ciarams87 ?

**Review by soneillf5 — APPROVED** [review]

File: .github/actions/smoke-tests/action.yaml (hunk 1 above)
Comment (lucacome) [reply .github/actions/smoke-tests/action.yaml:67]: `inputs.nginx-crt` and key are taking the input value from https://github.com/nginxinc/kubernetes-ingress/pull/2189/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR178 and there are stored as secrets in GitHub like all the other secrets. Does that answer your question? I wasn't really sure which part you were asking about

File: .github/actions/smoke-tests/action.yaml (hunk 2 above)
Comment (lucacome) [reply .github/actions/smoke-tests/action.yaml:45]: Just to clarify this is just to figure out which type of tests to run based on the name of the image. All the plus images have `plus` in the name.

Comment (ciarams87) [reply .github/actions/smoke-tests/action.yaml:45]: Other than hardcoding it elsewhere, there isn't many other good options. The test suite literally does the same thing when judging whether or not it should run the AP tests. I think this is fine.

**Review by ciarams87 — APPROVED** [review]

**Pull request merged**